--- a/src/spatialite/spatialite.c	2011-08-17 17:50:36.000000000 +0200
+++ b/src/spatialite/spatialite.c	2011-10-18 12:15:24.707185413 +0200
@@ -17512,6 +17512,7 @@
 /* setting a timeout handler */
     sqlite3_busy_timeout (db, 5000);
 
+#if 0
     printf ("SpatiaLite version ..: %s", spatialite_version ());
     printf ("\tSupported Extensions:\n");
 #ifndef OMIT_ICONV		/* ICONV is required by SHP/DBF/TXT */
@@ -17537,6 +17538,7 @@
     printf ("GEOS version %s\n", GEOSversion ());
     fflush (stdout);
 #endif /* end GEOS version */
+#endif
     return 0;
 }
