--- a/nvidia_bl.c	2010-11-04 16:05:05.000000000 +0100
+++ b/nvidia_bl.c	2011-04-02 14:14:30.501005970 +0200
@@ -513,6 +513,9 @@
 	/* Quadro FX 1800M (195.36) */
 	{ PCI_VDEVICE(NVIDIA, 0x0cbc), (kernel_ulong_t)&nv5x_driver_data },

+	/* Geforce GT 330M (260.56) */
+	{ PCI_VDEVICE(NVIDIA, 0x0df0), (kernel_ulong_t)&nv5x_driver_data },
+
 	/* end of list */
 	{ }
 };

@@ -775,6 +775,15 @@
 		},
 		.driver_data	= (void *)&max_level_0x1ffff,
 	},
+	{
+		.callback	= &nvidia_bl_dmi_match,
+		.ident		= "Sony VPCF13C5E",
+		.matches	= {
+			DMI_MATCH(DMI_SYS_VENDOR, "Sony Corporation"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "VPCF13C5E"),
+		},
+		.driver_data	= NULL,
+	},

 	/*
 	 * Dell machines: Nvidia graphics
