--- spatialite.c	2009-06-29 18:18:15.000000000 +0200
+++ spatialite_new.c	2009-12-16 17:21:09.000000000 +0100
@@ -29226,6 +29226,7 @@
 /* setting a timeout handler */
     sqlite3_busy_timeout (db, 5000);
 
+#ifdef FORCE_VERBOSE
     printf ("SpatiaLite version ..: %s", spatialite_version ());
     printf ("\tSupported Extensions:\n");
     printf ("\t- 'VirtualShape'\t[direct Shapefile access]\n");
@@ -29243,6 +29244,7 @@
     printf ("GEOS version %s\n", GEOSversion ());
     fflush (stdout);
 #endif /* end GEOS version */
+#endif
     return 0;
 }

